### ISSUE TITLE
Bug: Indentation parser state leaks between files, causing order-dependent false positives

### ISSUE THREAD

**Issue by grzegorzkoper** [aristanetworks/j2lint#292]
In j2lint, the jinja-statements-indentation rule can retain parser state from one file and apply it to the next file.
j2lint/linter/indenter/node.py defines these as module-level globals:
```
jinja_node_stack = []
jinja_delimiter_stack = []
```
JinjaTemplateIndentationRule.checktext() creates a new root Node, but does not clear those stacks before or after processing a file.

If parsing raises JinjaLinterError or IndexError, checktext() catches the exception. Any entries left on the global stacks remain present when the next file is checked. 
This causes valid files to fail depending on input order.

Minimal reproducer

contaminator.j2:
```
{% if enabled %}{% for item in items %}
{{ item }}{% endfor %}{% endif %}
```
victim.j2:
```
text{% if enabled %}
value
{% endif %}
```
Each file passes independently:
```
j2lint contaminator.j2 --ignore single-statement-per-line
# exit 0

j2lint victim.j2 --ignore single-statement-per-line
# exit 0
```
Together, with a deterministic hash seed:
```
PYTHONHASHSEED=0 j2lint \
    contaminator.j2 victim.j2 \
    --ignore single-statement-per-line
```
Result:
```
victim.j2:3
Tag is out of order 'endif' (jinja-statements-indentation)

Jinja2 linting finished with 1 error(s)

Exit code: 2.
```
Reversing the effective file order passes.

Expected result:

Each file should be parsed with fresh indentation state. Lint results should not depend on other input files or traversal order.

Actual result:

An internal parse failure can leave entries in jinja_node_stack. A later file is then evaluated against nodes belonging to a previous file and may receive a false error.

Additional nondeterminism:
It was randomly triggering in our CI in files unrelated to current patchsets, turns out the CLI converts positional files through a set, and recursive file discovery uses unsorted os.walk() results. This makes the effective processing order platform and run dependent.
